### ISSUE TITLE
Support JWTCreator.Builder to accept Json values

### ISSUE THREAD

**Issue by andrewrigas** [auth0/java-jwt#622]
I am trying to write a Scala library which basically enhance the functionality on top of [auth0/java-jwt](https://github.com/auth0/java-jwt) library. The problem I need to address is to be able to convert scala types via the encoders in java types, which is not really easy to do because you can have nested types. It will be nice to have an api that would accept a json directly and make some checks that the json passed was in a valid structure for payload and header.

Ideal solution is to be able to expose those 2 functions illustrated below:

```
public Builder withHeader(String headerClaimsJson) 

public Builder withPayload(String payloadClaimsJson)
```

**Comment by jimmyjames** [comment]
:wave: @andrewrigas - I don't think this is something we can reasonably support, as we need to ensure the types/values are of our supported serializable types, and then convert them to a map of string->object. Parsing apart raw JSON strings is something that the client can do and then use existing APIs including the `withPayload(Map<String, ?> payloadClaims)` method.

**Comment by andrewrigas** [comment]
@jimmyjames - Not sure exactly why do you need to convert it back to `Map<String, Object>`? If for example a client passed a valid JSON string structure, then you could convert the rest of the claims to JSON string and merge. The problem with passing a `Map` into the `withPayload(Map<String,?> payloadClaims)` is that I need to provide the map with a key. 

Therefore, the JSON will look like this:
```json 
 {
   "issuer": "foo",
   "subject": "bar",
   "key": {
       "name": "Sam",
       "age": 18
   }
 }
```

Where what I want to achieve is having the client claims in the payload like the example below:
```json
{
   "issuer": "foo",
   "subject": "bar",
   "name": "Sam",
   "age": 18
}
```

**Comment by jimmyjames** [comment]
> Where what I want to achieve is having the client claims in the payload like the example below:

I may be missing something - why can't you just use `name` and `age` as first-level keys in the map passed to `withPayload(Map<String,?> payloadClaims)`?

**Comment by jimmyjames** [comment]
Would something like this work? I think we could add a new method that accepts a JSON string representing the payload, and then delegate to the `withPayload(Map claims)` method to run the existing validation.

```java
public Builder withPayload(String payloadJSON) {
        try {
            Map<String, ?> claims = mapper.readValue(payloadJSON, Map.class);
            return withPayload(claims);
        } catch (JsonProcessingException e) {
            // TODO what exception to throw?
            throw new IllegalArgumentException("Invalid payload JSON", e);
        }
}
```

**Comment by andrewrigas** [comment]
I re-opened the PR: https://github.com/auth0/java-jwt/pull/643

**Comment by jimmyjames** [comment]
Fixed with #643 - we're targeting a new release with this change by end of week.
